x/poolmanager/router.go

```diff
@@ -545,6 +545,38 @@ func (k Keeper) AllPools(
 	return sortedPools, nil
 }
 
+// ListPoolsByDenom returns all pools by denom sorted by their ids
+// from every pool module registered in the
+// pool manager keeper.
+func (k Keeper) ListPoolsByDenom(
```

Comment:
Do you mind adding unit test for this?
After being added, we should be good to merge this

Reply:
Ok, I'll add test for listpoolsbydenom func